src/cards/split.ts

```diff
@@ -25,17 +25,16 @@ function remove(unit: Unit.IUnit, underworld: Underworld) {
   }
 
   const inverseMult = 1 / splitStatMultiplier;
-  for (let i = 0; i < modifier.quantity; i++) {
-    unit.healthMax = Math.max(1, Math.floor(unit.healthMax * inverseMult));
-    unit.health = Math.max(1, Math.floor(unit.health * inverseMult));
-    // Note: manaMax is not changed or else it would render casters useless
-    unit.mana = Math.floor(unit.mana * inverseMult);
-    unit.manaPerTurn = Math.floor(unit.manaPerTurn * inverseMult);
-    unit.staminaMax = Math.floor(unit.staminaMax * inverseMult);
-    unit.stamina = Math.floor(unit.stamina * inverseMult);
-    unit.damage = Math.floor(unit.damage * inverseMult);
-    unit.moveSpeed *= inverseMult;
-  }
+  const multiplier = Math.pow(inverseMult, modifier.quantity);
+  unit.healthMax = Math.max(1, unit.healthMax * multiplier);
+  unit.health = Math.max(1, unit.health * multiplier);
+  // Note: manaMax is not changed or else split would render casters useless
+  unit.mana = unit.mana * multiplier;
+  unit.manaPerTurn = unit.manaPerTurn * multiplier;
+  unit.staminaMax = unit.staminaMax * multiplier;
+  unit.stamina = unit.stamina * multiplier;
+  unit.damage = unit.damage * multiplier;
+  unit.moveSpeed *= multiplier;
```

Comment:
This probably needs some manual testing, I remember we had issues with split in the past when it was added and removed

Reply:
I think the major issue was rounding errors:

Golem hp during split: 30 -> 15 -> 7 > 3
Golem hp during purify: 3 -> 6 -> 12 -> 24

We had to use a for loop to simulate each step of "rounding" applied by each quantity of split, but since it doesn't round between steps anymore, we can just get the total multiplier and apply it immediately. I did a quick test on a couple of archers, and they seem to split/purify as expected.

I didn't do thorough testing with other modifiers involved, though